ndonnx/_typed_array/typed_array.py

```diff
@@ -457,7 +457,58 @@ def variance(
     def repeat(
         self, repeats: int | TyArrayInt64, /, *, axis: int | None = None
     ) -> Self:
-        raise _make_type_error("repeat", self.dtype)
+        # Default implementation of `repeat` that does not depend on the underlying layout
+        from . import onnx
+
+        x = self
+        if axis is None:
+            x = x.reshape((-1,))
+            axis = 0
+
+        if axis < 0:
+            axis += x.ndim
+
+        x_shape = x.dynamic_shape
+
+        if isinstance(repeats, onnx.TyArrayInt64) and repeats.ndim > 1:
+            raise ValueError(
+                f"'repeats' must be 0 or 1 dimensional, but has `{repeats.ndim}` dimensions"
+            )
+
+        if isinstance(repeats, int):
+            repeats = onnx.const(np.asarray(repeats), onnx.int64)
+
+        if repeats.ndim == 0:
+            repeats = repeats[None]
+            # Expand by one dimension and use regular broadcasting to repeat
+            key: list[slice | None] = [slice(None, None)] * (x.ndim + 1)
+            key[axis + 1] = None
+            tmp = x[tuple(key)].broadcast_to(
+                x_shape[: axis + 1].concat([repeats, x_shape[axis + 1 :]], axis=0)
+            )
+            # We now have a shape of (..., self.shape[axis], repeats, ...).
+            # All we need to do is collapse these two axes into one.
+            out_shape = x_shape.copy()
+            out_shape[axis] = safe_cast(onnx.TyArrayInt64, out_shape[axis] * repeats)
+
+            return tmp.reshape(out_shape)
```

Comment:
```suggestion
            x_shape[axis] = x_shape[axis] * repeats
            return tmp.reshape(x_shape)
```

It'd be better if you updated the type hint in the typed integer array class for multiplication rather than do this cast.

Reply:
Fair enough. Let's do that next time!